File: LMIS/LMIS.Web/Modules/ServerTypes/ReaderManage/CardlevelsForm.ts
import { StringEditor, DecimalEditor, IntegerEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface CardLevelsForm {
    CardLevelName: StringEditor;
    Discount: DecimalEditor;
    Penalty: DecimalEditor;
    BorrowDays: IntegerEditor;
    BorrowBooks: IntegerEditor;
    CreateTime: DateEditor;
    UpdateTime: DateEditor;
}

export class CardLevelsForm extends PrefixedContext {
    static readonly formKey = 'ReaderManage.CardLevels';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CardLevelsForm.init)  {
            CardLevelsForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = IntegerEditor;
            var w3 = DateEditor;

            initFormType(CardLevelsForm, [
                'CardLevelName', w0,
                'Discount', w1,
                'Penalty', w1,
                'BorrowDays', w2,
                'BorrowBooks', w2,
                'CreateTime', w3,
                'UpdateTime', w3
            ]);
        }
    }
}

File: LMIS/LMIS.Web/Modules/ReaderManage/CardLevels/CardLevelsForm.cs
﻿using Serenity.ComponentModel;
using System;

namespace LMIS.ReaderManage.Forms;

[FormScript("ReaderManage.CardLevels")]
[BasedOnRow(typeof(CardLevelsRow), CheckNames = true)]
public class CardLevelsForm
{
    public string CardLevelName { get; set; }
    public decimal Discount { get; set; }
    public decimal Penalty { get; set; }
    public int BorrowDays { get; set; }
    public int BorrowBooks { get; set; }
    public DateTime CreateTime { get; set; }
    public DateTime UpdateTime { get; set; }
}